zebra/zebra_rib.c

```diff
@@ -491,6 +513,18 @@ nexthop_active_ipv4 (struct rib *rib, struct nexthop *nexthop, int set,
 				  resolved_hop->flags |= NEXTHOP_FLAG_ONLINK;
 			      }
 			  }
+			if (newhop->type == NEXTHOP_TYPE_IPV6
```

Comment:
can you go the extra mile and split the common code from `rib->type == ZEBRA_ROUTE_STATIC` and `CHECK_FLAG (rib->flags, ZEBRA_FLAG_INTERNAL)` into a separate function? It's _almost_ identical...